packages/jsii-pacmak/lib/targets/dotnet/dotnetdocgenerator.ts

```diff
@@ -156,7 +161,11 @@ export class DotNetDocGenerator {
   }
 
   private convertExample(example: string): string {
-    const snippet = typeScriptSnippetFromSource(example, 'example');
+    const snippet = typeScriptSnippetFromSource(
+      example,
+      'example',
+      enforcesStrictMode(this.assembly),
```

Comment:
Why have you chosen for the DocGenerator classes to now depend on the jsii assembly? Is this intentional?

Can be avoided if the `enforceStrictAssembly` is called at the layer above. This will also avoid code duplication across the different language targets.

Reply:
`jsii-pacmak` right now uses a single Rosetta instance for all packages it generates (there could be multiple from `--recurse`). Creating multiple instances of Rosetta means creating multiple TypeScript contexts, which I decided against.